pandas/tests/window/test_rolling.py

```diff
@@ -96,71 +90,33 @@ def test_invalid_constructor(frame_or_series, w):
         timedelta(days=3),
         Timedelta(days=3),
         "3D",
-        ExpandingIndexer(window_size=3),
-        ExponentialMovingWindowIndexer(window_size=3),
-        GroupbyIndexer(window_size=3),
         VariableOffsetWindowIndexer(
             index=date_range("2015-12-25", periods=5), offset=BusinessDay(1)
         ),
-        VariableWindowIndexer(window_size=3),
-    ],
-)
-@pytest.mark.parametrize(
-    "func",
-    [
-        lambda df: df.rolling,
-        lambda df: df.groupby("key").rolling,
     ],
 )
-def test_constructor_step_not_implemented(window, func, step):
+def test_freq_window_not_implemented(window):
     # GH 15354
     df = DataFrame(
-        {"value": np.arange(10), "key": np.array([1] * 5 + [2] * 5)},
+        np.arange(10),
         index=date_range("2015-12-24", periods=10, freq="D"),
     )
-    f = lambda: func(df)(window=window, step=step)
-    if step is None:
-        f()
-    else:
-        with pytest.raises(NotImplementedError, match="step not implemented"):
-            f()
+    with pytest.raises(
+        NotImplementedError, match="step is not supported with frequency windows"
+    ):
+        df.rolling("3D", step=3)
 
 
 @pytest.mark.parametrize("agg", ["cov", "corr"])
-def test_constructor_step_not_implemented_for_cov_corr(agg, step):
-    # GH 15354
-    df = DataFrame(
-        {"value": np.arange(10), "key": np.array([1] * 5 + [2] * 5)},
-        index=date_range("2015-12-24", periods=10, freq="D"),
-    )
-    f = lambda: getattr(df.rolling(window=2, step=step), agg)(df)
-    if step is None:
-        f()
-    else:
-        with pytest.raises(NotImplementedError, match="step not implemented"):
-            f()
-
-
-@pytest.mark.parametrize(
-    "func",
-    [
-        lambda df: df.expanding,
-        lambda df: df.ewm,
-    ],
-)
-def test_constructor_step_unsupported(func, step):
+def test_step_not_implemented_for_cov_corr(agg):
```

Comment:
do we have another test which asserts NotImplemented for freq based step?

Reply:
`test_constructor_step_not_implemented` has the `window` parameter tested for "3D". Is this the freq test you're looking for?

Reply:
Yup, exactly (which was renamed `test_freq_window_not_implemented`)